### PR TITLE
Feature/dense submap matching

### PR DESCRIPTION
**Pull request by patripfr**
this adds incremental dense submap matching, pointcloud to submap matching, flag to ignore certain pointcloud fields
do not merge yet

### CODE REVIEW

File: common/resources-common/include/resources-common/point-cloud.h [hunk 1]
```diff
@@ -34,7 +34,7 @@ struct PointCloud {
   std::vector<uint32_t> labels;
   std::vector<uint16_t> rings;
   std::vector<float> times;
-
+  bool is_undistorted = false;
```
Comment (patripfr) [inline common/resources-common/include/resources-common/point-cloud.h:37]: rm

File: common/resources-common/src/point-cloud.cc [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 #include <numeric>
 
+#include <aslam/common/pose-types.h>
```
Comment (patripfr) [inline common/resources-common/src/point-cloud.cc:5]: rm

File: common/resources-common/src/point-cloud.cc [hunk 3]
```diff
@@ -282,4 +283,5 @@ bool PointCloud::loadFromCompressedFile(const std::string& file_path) {
   }
   return true;
 }
+
```
Comment (patripfr) [inline common/resources-common/src/point-cloud.cc:286]: rm

File: console-plugins/dense-mapping-constraints-plugin/src/dense-mapping-search.cc [hunk 4]
```diff
@@ -148,14 +160,34 @@ void findAllAlignmentCandidates(
...
       for (const pose_graph::VertexId& vertex_id : vertex_ids) {
-        const int64_t vertex_timestamp_ns =
-            map.getVertex(vertex_id).getMinTimestampNanoseconds();
+        const vi_map::Vertex& vertex = map.getVertex(vertex_id);
+        const int64_t vertex_timestamp_ns = vertex.getMinTimestampNanoseconds();
         temporal_vertex_buffer.addValue(vertex_timestamp_ns, vertex_id);
+        if (config.min_mission_distance_m > 0.0 &&
+            skip_timestamps_earlier_than_ns ==
+                std::numeric_limits<int64_t>::max()) {
+          // LOG(WARNING) << (p_M_I_first_vertex - vertex.get_p_M_I()).norm();
```
Comment (patripfr) [inline console-plugins/dense-mapping-constraints-plugin/src/dense-mapping-search.cc:184]: rm

**Review by LBern — APPROVED** [review]
looking good.

**Pull request merged**